src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -237,8 +231,7 @@ int32_t SystemNative_GetHostEntryForName(const uint8_t* address, struct HostEntr
     entry->CanonicalName = NULL;
     entry->Aliases = NULL;
     entry->AddressListHandle = (void*)info;
```

Comment:
We can remove this cast too, right?

Reply:
yep i'm working on it in this moment...also change parameter of SystemNative_GetNextIPAddress to struct addrinfo** after that no more cast to void*

Reply:
great, thanks.

Reply:
do we need to change also signature on Interop.HostEntry.cs(adding struct addinfo)?

Reply:
I'm not sure if there's a general pattern for this sort of thing.

That said, what I'd suggest is to define `struct addrinfo` in managed code but leave it empty.  That way the type is defined and all the type checking is proper, but the struct is still opaque to managed code.

Reply:
Ok i'll work on it!

Reply:
done!